src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1400,6 +1405,11 @@ private void EnterParameters(ImmutableArray<ParameterSymbol> parameters)
 
         protected virtual void EnterParameter(ParameterSymbol parameter)
         {
+            if ((parameter.ContainingSymbol as MethodSymbol)?.MethodKind == MethodKind.LocalFunction)
```

Comment:
I do not see what can become out of sync here. I still think that the code belongs in VisitLocalFunctionStatement and that we shouldn't have a special case for a local function here. This is local function specific logic and to look for this logic I would go to VisitLocalFunctionStatement.